### PR TITLE
Adding a possibility to return all env variables without prefix

### PR DESCRIPTION
**Pull request by e-ndrus**
Hello @morficus. Thanks for this great project. Submitting a PR for the plugin to see all env variables, not just those prefixed with CYPRESS_. Also adding a test for that functionality.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,4 +42,8 @@ module.exports = (on, config) => {
 ```
 
 ## Options
-This plugin takes two paramaters. The first parameter (which is mandatory) is the Cypress config object and the other is an optional [dotenv](https://www.npmjs.com/package/dotenv#config) config object.
+This plugin takes three paramaters. The first parameter (which is mandatory) is the Cypress config object. 
+
+The second is an optional [dotenv](https://www.npmjs.com/package/dotenv#config) config object.
+
+The third is an optional [all] parameter, which is set to false by default. If set to true, it returns all available environmental variables, not limited to those prefixed with CYPRESS_.
```
Comment (morficus) [inline README.md]: thank you for updating the documentation too! 🙏

File: index.js [hunk 2]
```diff
@@ -6,9 +6,10 @@ const clonedeep = require('lodash.clonedeep')
  *
  * @param {object} cypressConfig - The cypress config object
  * @param {object} dotEnvConfig - (optional) The dotenv config object, ref: https://www.npmjs.com/package/dotenv#config
+ * @param {string} all - (optional) Whether to return all env variables. If set to false (default), only env variables prefixed with CYPRESS_ are returned.
  * @returns {object} The cypress config with an augmented `env` property
  */
-module.exports = (cypressConfig, dotEnvConfig) => {
+module.exports = (cypressConfig, dotEnvConfig, all = false) => {
```
Comment (morficus) [inline index.js:12]: love this new parameter! that way it keeps the existing behavior and won't break stuff for others.

File: index.js [hunk 3]
```diff
@@ -6,9 +6,10 @@ const clonedeep = require('lodash.clonedeep')
  *
  * @param {object} cypressConfig - The cypress config object
  * @param {object} dotEnvConfig - (optional) The dotenv config object, ref: https://www.npmjs.com/package/dotenv#config
+ * @param {string} all - (optional) Whether to return all env variables. If set to false (default), only env variables prefixed with CYPRESS_ are returned.
```
Comment (morficus) [inline index.js]: I think this is a typo? since the 3rd parameter is a boolean and not a string

File: index.js [hunk 4]
```diff
@@ -18,15 +19,15 @@ module.exports = (cypressConfig, dotEnvConfig) => {
   enhancedConfig.env = enhancedConfig.env || {}
 
   // get the name of all env vars that relate to cypress
-  const cypressEnvVarKeys = Object.keys(envVars).filter(envName => envName.startsWith('CYPRESS_'))
+  const cypressEnvVarKeys = all
+    ? Object.keys(envVars).filter(envName => envName)
```
Comment (morficus) [inline index.js]: since nothing needs to be filtered out here, this could be simplified to just be `envName`

File: README.md (hunk 1 above)
Comment (morficus) [inline README.md]: Would you mind adding in here that the 3rd parameter is a boolean, please?

**Review by morficus — CHANGES_REQUESTED** [review]
@e-ndrus  - minor comments on three things that could be cleaned up before merging this in.
Thank you so much for the new feature! 🙏

File: index.js (hunk 4 above)
Comment (e-ndrus) [reply index.js]: @morficus oh, sorry, missed that. Updated the PR.

File: README.md (hunk 1 above)
Comment (e-ndrus) [reply README.md]: Updated.

File: index.js (hunk 3 above)
Comment (e-ndrus) [reply index.js]: Corrected.

**Review by morficus — APPROVED** [review]

**Comment by dandv** [comment]
Does this need some extra work?

1. In the README, the example talks only about the CYPRESS_ variable: `You can use Cypress.env('HELLO) [sic] to access its value.`
2. The test .env* files don't contain unprefixed variables.

**Comment by morficus** [comment]
@dandv check out [this unit test] (https://github.com/morficus/cypress-dotenv/blob/a486a1857f6e324f8e7e2d5e7fa050291d0ad985/index.spec.js#L106). This is how I confirmed the optional 3rd argument was working as expected.

**Pull request merged**